dotcom-rendering/src/web/components/Snap.tsx

```diff
@@ -14,15 +15,25 @@ export const Snap = ({ snapData }: Props) => {
 		return <></>;
 	}
 
+	// Emotion doesn't parse this conditional CSS correctly which is used by some thrashers to check
+	// if theres a pageskin. This regex fixes the vast majority of these cases should allow future snaps
+	// to be built with DCR in mind.
+	if (snapData.embedCss !== undefined && snapData.embedCss !== null) {
+		snapData.embedCss = snapData.embedCss.replace(
+			/body:not\(\.has-active-pageskin\)(?! &)/g,
+			'body:not(.has-active-pageskin) &',
+		);
+	}
```

Comment:
I would like this to be extracted to its own method, so it's easier to refactor/remove in the future.

I'm puzzled as whether the `embedCss` can be missing. Here we're checking it is defined, but below we removed the optional chaining…

Reply:
I'd argue it'd be good to keep here so it's in context, and isn't a bit of functionality that we would want to leak out into anywhere else in the code. 

How would making it a function make it easier to work with @mxdvl?

Reply:
We typically make model changes in enhancers. It's hard to know how to classify what css mutations are (because I don't think I've ever seen one happen before) but if we decide it's the same as the other enhancements, and we decide to keep it  (I'm still iffy on if we want this pattern or not) then maybe want to keep all of these types of changes in one place, in the `enhanceCollections` step?